### PR TITLE
chore(cli): warn if unsupported/missing typescript version

### LINKED ISSUE #5728: Add TypeScript version check
We need at least TypeScript 4.1 to run `@prisma/client`. This should be enforced to prevent bad surprises.

### PR DESCRIPTION
**Pull request by williamluke4**
closes https://github.com/prisma/prisma/issues/5728

When `tsc` is not present
Clipboard_2021-03-11-17-22-50
When `tsc` is outdated 
Clipboard_2021-03-11-17-25-13
I also added `typescript:>=4.1` to the cli's peerDeps as this will warn on install.

### CODE REVIEW

**Comment by Jolg42** [comment]
We should not require typescript dependency or warn that it's not there in my opinion.

If you're a Node.js user, not using TypeScript you don't want warnings about it if you're happy with your .js code.

**Comment by pantharshit00** [comment]
I agree. Warning when typescript is installed should be enough.

File: src/packages/sdk/src/predefinedGeneratorResolvers.ts [hunk 1]
```diff
@@ -177,13 +177,7 @@ function checkTypeScriptVersion() {
...
     }
   } catch (e) {
-    logger.error(
-      `You do not have ${chalk.bold(
-        'typescript',
-      )} installed. Please install at least version ${chalk.bold(
-        minVersion,
-      )} or ${chalk.bold('newer')} in order to use Prisma.`,
-    )
+    // They do not have TS installed to just pass
```
Comment (Jolg42) [inline src/packages/sdk/src/predefinedGeneratorResolvers.ts:180]:
```suggestion
    // They do not have TS installed, we ignore (example: JS project)
```

File: src/packages/sdk/src/predefinedGeneratorResolvers.ts [hunk 2]
```diff
@@ -155,7 +155,31 @@ function checkYarnVersion() {
...
+    })
+    if (output.stdout) {
+      const currentVersion = output.stdout.split(' ')[1]
+      if (semverLt(currentVersion, minVersion)) {
+        logger.warn(
+          `Your ${chalk.bold(
+            'typescipt',
+          )} version ${currentVersion}, which is outdated. Please update it to ${chalk.bold(
+            minVersion,
+          )} or ${chalk.bold('newer')} in order to use Prisma.`,
```
Comment (Jolg42) [inline src/packages/sdk/src/predefinedGeneratorResolvers.ts:175]:
```suggestion
          `Your ${chalk.bold(
            'typescript',
          )} version is ${currentVersion}, which is outdated. Please update it to ${chalk.bold(
            minVersion,
          )} or ${chalk.bold('newer')} in order to use Prisma Client.`,
```

**Pull request merged**